cdisc_rules_engine/services/data_readers/dataset_ndjson_reader.py

```diff
@@ -16,16 +16,39 @@
 
 
 class DatasetNDJSONReader(DataReaderInterface):
+    def __init__(self, dataset_implementation, encoding: str = None):
+        self.dataset_implementation = dataset_implementation
+        self.encoding = encoding
+
     def get_schema(self) -> dict:
         schema = JSONReader().from_file(
             os.path.join("resources", "schema", "dataset-ndjson-schema.json")
         )
         return schema
 
     def read_json_file(self, file_path: str) -> dict:
-        with open(file_path, "r") as file:
-            lines = file.readlines()
-        return json.loads(lines[0]), [json.loads(line) for line in lines[1:]]
+        if self.encoding:
+            with open(file_path, "r", encoding=self.encoding) as file:
+                lines = file.readlines()
+            return json.loads(lines[0]), [json.loads(line) for line in lines[1:]]
+
+        encodings_to_try = ["utf-8", "utf-16", "utf-32"]
```

Comment:
Should this be done by a user? I.e. a default encoding should be UTF-8 and if it fails a user gets a message and either updates the dataset to use UTF-8 or changes the encoding parameter value.